File: codejam/21/0q/b-moons-and-umbrellas.py
#!/usr/bin/env python3


def min_cost(cj, jc, pattern):
    cost = {'CC': 0, 'JJ': 0, 'CJ': cj, 'JC': jc}
    return sum(cost[pattern.replace('?', '')[i:i+2]] for i in range(len(pattern)-1))


def main():
    for case in range(int(input())):
        rcj, rjc, pattern = input().split()
        answer = min_cost(int(rcj), int(rjc), pattern)
        print(f'Case #{case+1}: {answer}')


if __name__ == '__main__':
    main()

File: codejam/21/0q/b-moons-and-umbrellas.rb
#!/usr/bin/env ruby

gets.to_i.times do |test|
  cj, jc, pattern = gets.split
  ans = 0
  pattern.delete("?").split("").each_cons(2) { |a,b|
    ans += cj.to_i if a+b == "CJ"
    ans += jc.to_i if a+b == "JC"
  }
  puts "Case ##{test+1}: #{ans}"
end
